dolphinscheduler-aop/src/main/java/org/apache/dolphinscheduler/aop/YarnClientAspect.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.dolphinscheduler.aop;
+
+import org.apache.hadoop.yarn.api.records.ApplicationId;
+import org.apache.hadoop.yarn.api.records.ApplicationReport;
+import org.apache.hadoop.yarn.api.records.ApplicationSubmissionContext;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.nio.file.StandardOpenOption;
+import java.util.Collections;
+
+import org.aspectj.lang.annotation.AfterReturning;
+import org.aspectj.lang.annotation.Aspect;
+
+@Aspect
+public class YarnClientAspect {
+
+    /**
+     * flag to indicate whether print debug logs
+     */
+    private static final String PARA_NAME_ASPECTJ_DEBUG = "PARA_NAME_ASPECTJ_DEBUG";
+
+    /**
+     * The current application report when application submitted successfully
+     */
+    private ApplicationReport currentApplicationReport = null;
+
+    private String appInfoFilePath;
+    private boolean debug;
+
+    public YarnClientAspect() {
+        appInfoFilePath = System.getProperty("user.dir") + "/appInfo.log";
+        debug = Boolean.parseBoolean(System.getenv(PARA_NAME_ASPECTJ_DEBUG));
```

Comment:
should we put this variable ```PARA_NAME_ASPECTJ_DEBUG``` into common.properties?

Reply:
> should we put this variable `PARA_NAME_ASPECTJ_DEBUG` into common.properties?

AFAIK, `PARA_NAME_ASPECTJ_DEBUG` can not be read if put in common.properties, because task and aop code runs in another JVM and different JVM can not share properties in normal way.

Reply:
if he depends on the common module, it can be solved. I think this variable must be configurable and cannot be configured by modifying the code

Reply:
> if he depends on the common module, it can be solved. I think this variable must be configurable and cannot be configured by modifying the code. if above cann't fits your needs.We should prioritize using the configuration file in this module. Welcome to give your feedback any comments



> if he depends on the common module, it can be solved. I think this variable must be configurable and cannot be configured by modifying the code. if above cann't fits your needs.We should prioritize using the configuration file in this module. Welcome to give your feedback any comments

`PARA_NAME_ASPECTJ_DEBUG` is now configured in `dolphinscheduler_env.sh`. Actually, i'm considering about the configuration `appId.file.path` which i've already added into common.properties(default value is '/appInfo.log'), however, it's hard coded now in the above aop code. I'll try to optimize this part of code.